docs/source/getting_started.rst

```diff
@@ -182,6 +183,9 @@ environment does not include ``zstash``.
 Instead, the developer will ``pip install .`` to build ``zstash`` with changes
 (see step 6 below).
 
+Furthermore, the dev environment includes QA tools such as code formatters, linters, and ``pre-commit``.
+**NOTE**: These QA tools are enforced using ``pre-commit`` checks in the CI/CD build, so you must use the dev environment for all contributions.
```

Comment:
It might be a good idea to spell out "quality assurance (QA)", "continuous integration/continuous delivery (CI/CD)" on the first use.